dataland-frontend/src/utils/DataTableDisplay.ts

```diff
@@ -43,3 +43,83 @@ export type ReportingPeriodOfDataSetWithId = {
   dataId: string;
   reportingPeriod: string;
 };
+
+/**
+ * Adds click event listeners on DataTable row headers to expand and collapse row
+ * @param expandedRowsOnClick function passes the latest list of expanded row id's
+ * @param newExpandedRowsCallback function that returns the updated (after click) list of expanded row id's
+ * @returns the map of rows and their click handlers needed for unmounting
+ */
+export function mountRowHeaderClickEventListeners(
+  expandedRowsOnClick: () => string[],
+  newExpandedRowsCallback: (newExpandedRows: string[]) => void,
+): Map<Element, EventListener> {
+  const handlerMap: Map<Element | HTMLButtonElement, EventListener> = new Map();
+  let expandedRowGroups: string[] = [];
+
+  const t = setTimeout(() => {
+    const buttonAttributeName = "data-parent-id";
+    const rows = Array.from(document.querySelectorAll("[data-row-header-click]")).filter((el) => el);
+    const rowButtons = rows
+      .map((el: Element) => {
+        const button = el.parentNode?.querySelector('button[data-pc-section="rowgrouptoggler"]');
+        if (button) {
+          button?.setAttribute(buttonAttributeName, el.id);
+          return button;
+        }
+        return void 0;
+      })
+      .filter((button): button is HTMLButtonElement => !!button);
+
+    [...rows, ...rowButtons].forEach((el: Element | HTMLButtonElement) => {
+      let clickHandler: EventListener | null = (evt): void => {
+        if (!el.id) {
+          evt.stopImmediatePropagation();
+        }
+
+        expandedRowGroups = expandedRowsOnClick();
+        if (!expandedRowGroups.includes(el.id)) {
+          expandedRowGroups.push(el.id);
+        } else {
+          expandedRowGroups = expandedRowGroups.filter((id: string) => id !== el.id);
+        }
+        newExpandedRowsCallback(expandedRowGroups);
+      };
+
+      let target;
+      if (el?.getAttribute(buttonAttributeName)) {
+        target = el;
+      } else {
+        target = el.parentNode;
+      }
+
+      if (target) {
+        handlerMap.set(target as Element, clickHandler);
+        target?.addEventListener("click", clickHandler);
```

Comment:
"?" is not needed since already checked if it exists

Reply:
Done.

Reply:
👍